bql/planner/data_access.go

```diff
@@ -127,6 +127,7 @@ func simpleExist(ctx context.Context, gs []storage.Graph, cls *semantic.GraphCla
 func simpleFetch(ctx context.Context, gs []storage.Graph, cls *semantic.GraphClause, lo *storage.LookupOptions, stmLimit int64, chanSize int, w io.Writer) (*table.Table, error) {
 	s, p, o := cls.S, cls.P, cls.O
 	lo = updateTimeBounds(lo, cls)
+	loStr := lo.String()
```

Comment:
I am not sure if caching lo.String to a variable really brings advantages

Reply:
The reason here to cache the `lo.String()` to a `string` variable is a bit different, I tried to explain it in the commit messages as well: the `lo` above is a **pointer** to a `storage.LookupOptions` struct, and this struct referenced by the pointer is changed multiple times in code (when setting global time bounds, when adding or removing `FilterOptions` to/from it, and so on). This way, depending on the time the closure is called in the tracer we may see a different `lo.String()` in the tracing output compared to the one we really wanted to trace at the moment of the `.Trace` call.

I found this problem when reviewing some traces and found an inconsistency with the `"Setting global lookup options to %s"` tracing message (it was showing a non-nil `FilterOptions` there, but at the moment of setting global lookup options we do not have any `FilterOptions` set yet!). This was a bit subtle to debug, and the way to correct it was just to pass `lo.String()` to a local variable before sending it to the tracer.

To avoid this problem happens again, I decided to repeat this process for all places in code on which the `lo` pointer was being captured by a tracing closure (commit `7a148a1`).